doc/nrf/app_dev/device_guides/nrf54h/ug_nrf54h20_suit_recovery.rst

```diff
@@ -236,3 +236,85 @@ To turn an application into a recovery application, the following steps have to
     The value of ``SUIT_RECOVERY_APPLICATION_PATH`` can contain variables like ``${ZEPHYR_NRF_MODULE_DIR}``
 
 #. When building the main application, set ``SB_CONFIG_SUIT_RECOVERY_APPLICATION_CUSTOM`` (or the Kconfig option name if a different one was chosen) to ``y``.
+
+Entering recovery via button press
+**********************************
+
+It is possible to enable a feature that allows the device to enter recovery mode by pressing a button.
+This can be used for several purposes, such as:
+
+* Recovering the device if the main firmware crashes and no JLink connection is available.
+* Using the recovery application as a "foreground update" application, allowing to save some space in the main application.
+* Debugging the recovery application.
+
+When this feature is enabled, the device will enter recovery mode if the button is pressed during boot.
+
+.. note::
+   To exit the recovery mode, a device firmware update must be performed.
+
+There are two variants of this feature:
+
+* Recovery button checked in the recovery application run as a companion image at boot stage (recommended).
+* Recovery button checked in the main application.
+
+The first variant is recommended as it allows to perform the check independently of the main application.
+
+Recovery button checked in the recovery application
+===================================================
+
+In this variant the recovery application is run as a companion image before the main application when the device boots.
+It checks if the device already is in recovery mode.
+If it is, the recovery application proceeds with its normal operation allowing to recover firmware.
```

Comment:
```suggestion
If so, the recovery application proceeds with its normal operation allowing to recover firmware.
```

Reply:
I think I'll let the tech writers decide on this one, it might slightly change the meaning

Reply:
IMO, "If so" sounds more accurate.

"_If so, the recovery application proceeds with its normal operation, allowing the firmware to be recovered._"?